c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -482,6 +482,32 @@ class CachingAllocatorConfig {
   }
 };
 
+// records allocate/free events
+class MemoryEventTracker {
+ private:
+  std::mutex mutex;
+
+ public:
+  std::vector<std::vector<AllocFreeEvent>> alloc_free_events;
+  void append_alloc_free_event(
+      intptr_t ptr,
+      size_t size,
+      bool type,
```

Comment:
nit: if we only care about allocate/free, we can save a type arg by converting size into a sized integer. 

This is also what profiler does: https://github.com/pytorch/pytorch/blob/6d23ef847e1c119939095ca1e8ce195e697b2a09/c10/cuda/CUDACachingAllocator.cpp#L794

Reply:
Done. Thank you.